nni/retiarii/execution/logical_optimizer/logical_plan.py

```diff
@@ -2,20 +2,23 @@
 # Licensed under the MIT license.
 
 import copy
-from typing import Dict, Tuple, Any, Union
+from typing import Dict, Tuple, Any
 
 from nni.retiarii.utils import uid
-from nni.common.device import GPUDevice
+from nni.common.device import Device
 
 from ...graph import Cell, Edge, Graph, Model, Node
 from ...operation import Operation, _IOPseudoOperation
 
 
-class CPUDevice:
+class CPUDevice(Device):
```

Comment:
it is a little strange that why not put `CPUDevice` into device.py? it should also be a `dataclass`?

Reply:
Good comment. I have moved `CPUDevice` in nni.common.device and mark it as a dataclass.